azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_stages_provisioning.py

```diff
@@ -44,3 +58,403 @@ def _run_op(self, op):
+                                stage_name=self.name,
+                                op_name=op.name,
+                                interval=retry_interval,
+                                error=error,
+                            )
+                        )
+
+                        initial_register_op.retry_after_timer = Timer(
+                            retry_interval, do_retry_after
+                        )
+                        initial_register_op.retry_after_timer.start()
+
+                    # Service success scenario
+                    else:
+                        registration_status = CommonProvisioningStage._get_registration_status(
+                            decoded_response
+                        )
+                        if registration_status == "assigned" or registration_status == "failed":
+                            # process complete response here
+                            # TODO Is there an response when service returns "failed" ?
+                            complete_registration_result = CommonProvisioningStage._form_complete_result(
+                                operation_id=operation_id,
+                                decoded_response=decoded_response,
+                                status=registration_status,
+                            )
+                            initial_register_op.registration_result = complete_registration_result
+                            if registration_status == "failed":
+                                error = exceptions.ServiceError(
+                                    "Registration operation returned failed registration status  with a status code of {status_code}".format(
+                                        status_code=success_status_code
+                                    )
+                                )
+                            # complete the original op in both cases
+                            initial_register_op.complete(error=error)
+                        elif registration_status == "assigning":
+
+                            def copy_result_to_original_op(op, error):
+                                logger.debug(
+                                    "Copying registration result from Query Status Op to Registration Op"
+                                )
+                                initial_register_op.registration_result = op.registration_result
+                                initial_register_op.error = error
+
+                            query_worker_op = initial_register_op.spawn_worker_op(
+                                worker_op_type=pipeline_ops_provisioning.SendQueryRequestOperation,
+                                request_payload=" ",
+                                operation_id=operation_id,
+                                callback=copy_result_to_original_op,
+                            )
+
+                            time.sleep(constant.DEFAULT_POLLING_INTERVAL)
```

Comment:
>me.sleep(constant.DEFAULT_POLLING_INTERVAL) [](start = 30, length = 43)

EEK!  No! You can't sleep inside the pipeline thread.  This needs to be triggered by some sort of timer.